include/range/v3/view/iota.hpp

```diff
@@ -336,36 +389,59 @@ namespace ranges
         private:
             friend range_access;
             From from_;
+            step step_;
 
             From read() const
             {
                 return from_;
             }
-            void next()
+            CPP_member
+            auto next() -> CPP_ret(void)( //
+                requires(!detail::iota_stridable_<From, To>))
             {
                 ++from_;
             }
-            bool equal(sentinel const & that) const
+            CPP_member
+            auto next() -> CPP_ret(void)( //
+                requires detail::iota_stridable_<From, To>)
+            {
+                detail::iota_advance_(from_, step_.value());
+            }
+            CPP_member
+            auto equal(sentinel const & that) const -> CPP_ret(bool)( //
+                requires(!detail::iota_stridable_<From, To>))
             {
                 return from_ == that.to_;
             }
             CPP_member
+            auto equal(sentinel const & that) const -> CPP_ret(bool)( //
+                requires detail::iota_stridable_<From, To>)
+            {
+                return detail::iota_distance_(from_, that.to_) >= detail::iota_abs_(step_.value());
```

Comment:
This function looks wrong to me and I suspect it's never getting called from the tests. 

The case where `step == 1` should collapse into the case where there is no step, since that's the default. The no step sentinel is `from == to`. But this case is `to - from >= 1` or `from <= to - 1`, which is a different condition.

Reply:
Yes, the `sentinel` case is not being tested.

Reply:
It can never happen, I removed the function entirely